test/controllers/tweets_controller_test.rb

```diff
@@ -1,14 +1,63 @@
+require 'minitest/autorun'
 require 'test_helper'
+# require 'date'
 
 class TweetsControllerTest < ActionController::TestCase
   include Devise::Test::ControllerHelpers
 
   def setup
     @lesson = lessons(:test_lesson)
+    @tweet = tweet(:test_first_tweet)
+    @request.env["devise.mapping"] = Devise.mappings[:admin]
```

Comment:
What does this line do? I haven't seen anything quite like it, so it might be worth a comment if it's essential to running these tests

Reply:
Good point!  I added a clarifying note about testing with devise, and you can read more in [Devise's documentation about testing controllers](https://github.com/plataformatec/devise#controller-tests) if you're interested